lib/esp32/espnow.toit

```diff
@@ -0,0 +1,79 @@
+// Copyright (C) 2022 Toitware ApS. All rights reserved.
+// Use of this source code is governed by an MIT-style license that can be
+// found in the lib/LICENSE file.
+
+STATION ::= 0 // Wi-Fi station mode
+SOFTAP  ::= 1 // Not support yet
+
+check_mac_ address/ByteArray:
+  if address.size != 6:
+      throw "ESP-Now MAC address length must be 6 bytes"
+
+check_key_ key/ByteArray:
+  if key.size != 16:
+      throw "ESP-Now key length must be 16 bytes"
+
+check_channel_ channel/int:
+  if not 0 <= channel <= 14:
+      throw "ESP-Now channel range must be 0-14"
+
+class Address:
+  mac/ByteArray
+
+  constructor .mac:
+    check_mac_ mac
+
+  stringify -> string:
+    return "$(%02x mac[0]):$(%02x mac[1]):$(%02x mac[2]):$(%02x mac[3]):$(%02x mac[4]):$(%02x mac[5])"
+
+class Key:
+  data/ByteArray
+
+  constructor .data/ByteArray:
+    check_key_ data
+  
+  constructor.from_string string_data/string:
+    return Key string_data.to_byte_array
+
+class Datagram:
+  address/Address
+  data/ByteArray
+
+  constructor .address .data:
+
+class Service:
+  _group ::= ?
+
+  constructor --mode/int=STATION:
+    _group = espnow_init_ mode #[]
+  
+  constructor.with_key --mode/int=STATION --key/Key:
+    _group = espnow_init_ mode key.data
```

Comment:
I think this is more convenient (and rename the `STATION` to `STATION_` in line 5).
```suggestion
  constructor.station --key/Key?:
    key_data = key ? key.data : #[]
    _group = espnow_init_ STATION_ key_data
```

Basically, the call would then be:
```
service := espnow.Service.station
// or
service := espnow.Service.station --key=(espnow.Key #[1, 2, 3])
```

If the `station` mode is used significantly more often, then we might not need a named constructor for the station mode:
```suggestion
  constructor --key/Key?:
    key_data = key ? key.data : #[]
    _group = espnow_init_ STATION_ key_data
```

Once we support SoftAP, we can then add a named constructor:
```
  constructor.soft_ap ...:
```

Alternatively, if you think the mode flag is just better, we can add the mode flag at that point.

Reply:
Yes, using this method will look more direct.